src/interpreter.rs

```diff
@@ -118,95 +109,124 @@ impl Interpreter<Event, Command> for EventInterpreter {
                 }
             }
 
+            Event::UptaneMetadataUpdated(snapshot, targets) => {
+                unimplemented!();
+            }
+
             _ => ()
         }
     }
 }
 
 
-/// The `CommandInterpreter` wraps each incoming `Command` inside an `Interpret`
-/// type with no response channel for sending to the `GlobalInterpreter`.
-pub struct CommandInterpreter;
+/// The `IntermediateInterpreter` wraps each incoming `Command` inside an
+/// `Interpret` type with no response channel for sending to the `CommandInterpreter`.
+pub struct IntermediateInterpreter;
 
-impl Interpreter<Command, Interpret> for CommandInterpreter {
+impl Interpreter<Command, Interpret> for IntermediateInterpreter {
     fn interpret(&mut self, cmd: Command, itx: &Sender<Interpret>) {
-        info!("CommandInterpreter received: {}", cmd);
+        info!("IntermediateInterpreter received: {}", cmd);
         itx.send(Interpret { command: cmd, response_tx: None });
     }
 }
 
 
-/// The `GlobalInterpreter` interprets the `Command` inside incoming `Interpret`
+/// The `CommandMode` toggles the `Command` handling procedure.
+pub enum CommandMode {
+    Rvi(Box<Services>),
+    Uptane(Option<Uptane>),
+    Sota
+}
+
+/// The `CommandInterpreter` interprets the `Command` inside incoming `Interpret`
 /// messages, broadcasting `Event`s globally and (optionally) sending the final
 /// outcome `Event` to the `Interpret` response channel.
-pub struct GlobalInterpreter {
-    pub config:      Config,
-    pub auth:        Auth,
-    pub http_client: Box<Client>,
-    pub rvi:         Option<Services>
+pub struct CommandInterpreter {
+    pub mode:   CommandMode,
+    pub config: Config,
+    pub auth:   Auth,
+    pub http:   Box<Client>
 }
 
-impl Interpreter<Interpret, Event> for GlobalInterpreter {
+impl Interpreter<Interpret, Event> for CommandInterpreter {
     fn interpret(&mut self, interpret: Interpret, etx: &Sender<Event>) {
-        info!("GlobalInterpreter received: {}", interpret.command);
-
-        let (multi_tx, multi_rx) = chan::async::<Event>();
+        info!("CommandInterpreter received: {}", interpret.command);
+        let (events_tx, events_rx) = chan::async::<Event>();
 
         let outcome = match self.auth {
-            Auth::None => match self.config.auth {
-                None    => self.authenticated(interpret.command, multi_tx),
-                Some(_) => self.unauthenticated(interpret.command, multi_tx),
```

Comment:
This should be handled by `config.auth_command()`, where `Auth::None` is only for when no authentication is needed.